tools/flaky-test-reporter/main.go

```diff
@@ -64,9 +75,15 @@ func main() {
 		repoDataAll = append(repoDataAll, rd)
 	}
 
-	ghi, err := Setup(*githubToken)
-	if err != nil {
-		log.Fatalf("Cannot setup github: %v", err)
+	githubErr := ghi.processGithubIssues(repoDataAll, *dryrun)
+	slackErr := sendSlackNotifications(repoDataAll, slackClient, ghi, *dryrun)
+	if nil != githubErr {
+		log.Printf("Github step failures:\n%v", githubErr)
+	}
+	if nil != slackErr {
+		log.Printf("Slack step failures:\n%v", slackErr)
+	}
+	if nil != githubErr || nil != slackErr { // Fail this job if there is any error
```

Comment:
can we combine this in the if condition above?

Reply:
do you mean `defer`?

Reply:
I mean, we dont need another if condition here. Maybe we can have a exit func and call that in the if conditions above?

func exit(msg string, args...) {
log.Printf(msg, args...)
os.Exit(1)
}

Reply:
there might be cases where there are errors caught from both github and slack, and they should both be printed at the end for easier debugging, it's not practical to exit in the first if condition